### PR TITLE
Kraken and Bittrex syncing bugs

### PR DESCRIPTION
**Pull request by einsteinx2**
**Is this pull request adding a feature or fixing a bug?**  
bug

**Does this pull request close an issue? If so, which one?**
#481

**What does this pull request do? What does it change?**
Fixes Kraken api syncing calls completion handler twice and Bittrex syncing errors

### CODE REVIEW

File: Balance/Shared/Data Model/APIs/BITTREX/BITTREXApi.swift [hunk 1]
```diff
@@ -54,10 +54,13 @@ class BITTREXApi: NewExchangeApi, ExchangeApi {
             }
             
             institution = existingInstitution ?? InstitutionRepository.si.institution(source: .bittrex, sourceInstitutionId: "", name: "Bittrex")
+            institution?.apiKey = key
```
Comment (wolffan) [inline Balance/Shared/Data Model/APIs/BITTREX/BITTREXApi.swift]: i'm pretty sure you don't need to do that as they are stored on the credentials object...

File: Balance/Shared/Data Model/APIs/BITTREX/BITTREXApi.swift [hunk 2]
```diff
@@ -94,8 +97,7 @@ class BITTREXApi: NewExchangeApi, ExchangeApi {
             return callResultTaskWithError(APIBasicError.incorrectLoginCredentials, completionBlock: completionBlock)
         }
         
-        let dataTask = urlSession.dataTask(with: request) { [weak self] (data, response, error) in
-            guard let `self` = self else { return }
```
Comment (wolffan) [inline Balance/Shared/Data Model/APIs/BITTREX/BITTREXApi.swift]: I think we need to talkd about weakself vs strongself in clojures in swift...

File: Balance/Shared/Data Model/Syncing/SyncManager.swift [hunk 3]
```diff
@@ -50,8 +50,8 @@ class SyncManager: NSObject {
     deinit {
         #if os(OSX)
             NSWorkspace.shared.notificationCenter.removeObserver(self, name: NSWorkspace.didWakeNotification, object: nil)
-            NotificationCenter.removeObserverOnMainThread(self, name: Notifications.PopoverWillShow)
-            NotificationCenter.removeObserverOnMainThread(self, name: Notifications.PopoverWillHide)
+            NotificationCenter.removeObserverOnMainThread(self, name: Notifications.PopoverDidShow)
```
Comment (wolffan) [inline Balance/Shared/Data Model/Syncing/SyncManager.swift]: this seems unrelated with the pr

File: Balance/Shared/Data Model/Syncing/Syncer.swift [hunk 4]
```diff
@@ -84,6 +85,14 @@ class Syncer {
                         }
                     }
                 }
+            } else if institution.source == .bittrex {
```
Comment (wolffan) [inline Balance/Shared/Data Model/Syncing/Syncer.swift]: i'm not sure this make sense here... does it even hit this if in this condition?
we are mixing the existing logic with the logic I tried to apply to polo

File: Balance/Shared/Data Model/APIs/BITTREX/BITTREXApi.swift (hunk 1 above)
Comment (einsteinx2) [reply Balance/Shared/Data Model/APIs/BITTREX/BITTREXApi.swift]: This bittrex implementation doesn't use one, it gets credentials passed in to it's functions. When we standardize the APIs we'll have them all use the same storage method and migrate existing data.

File: Balance/Shared/Data Model/APIs/BITTREX/BITTREXApi.swift (hunk 2 above)
Comment (einsteinx2) [reply Balance/Shared/Data Model/APIs/BITTREX/BITTREXApi.swift]: In this case we needed to specifically not use weak self so that the API object doesn't die while the API call is executing (this was causing a bug). And agreed, in most cases we don't need to and in fact shouldn't use weak self.

File: Balance/Shared/Data Model/Syncing/SyncManager.swift (hunk 3 above)
Comment (einsteinx2) [reply Balance/Shared/Data Model/Syncing/SyncManager.swift]: Sorry this fixes a second issue (it's in it's own commit). I just noticed it while working on this and forgot to make it a separate branch.

File: Balance/Shared/Data Model/Syncing/Syncer.swift (hunk 4 above)
Comment (einsteinx2) [reply Balance/Shared/Data Model/Syncing/Syncer.swift]: Yes this is necessary or it never runs (syncing bug I noticed). The whole syncer is being refactored asap to be much simpler, so these inconsistencies are going away soon.
